contract_manager/src/chains.ts

```diff
@@ -560,3 +562,70 @@ export class AptosChain extends Chain {
     return { id: result.hash, info: result };
   }
 }
+
+export class FuelChain extends Chain {
+  static type = "FuelChain";
+
+  constructor(
+    id: string,
+    mainnet: boolean,
+    wormholeChainName: string,
+    nativeToken: TokenId | undefined,
+    public gqlUrl: string
+  ) {
+    super(id, mainnet, wormholeChainName, nativeToken);
+  }
+
+  async getProvider(): Promise<Provider> {
+    return await Provider.create(this.gqlUrl);
+  }
+
+  async getWallet(privateKey: PrivateKey): Promise<WalletUnlocked> {
+    const provider = await this.getProvider();
+    return Wallet.fromPrivateKey(privateKey, provider);
+  }
+
+  /**
+   * Returns the payload for a governance contract upgrade instruction for contracts deployed on this chain
+   * @param digest hex string of the 32 byte digest for the new package without the 0x prefix
+   */
+  generateGovernanceUpgradePayload(digest: string): Buffer {
+    return new UpgradeContract256Bit(this.wormholeChainName, digest).encode();
+  }
```

Comment:
I think you can throw an error here because we don't have any upgrade yet. Or because it is probably like this maybe leave a comment about it?